post-processor/docker-push/post-processor.go

```diff
@@ -104,13 +104,10 @@ func (p *PostProcessor) PostProcess(ctx context.Context, ui packer.Ui, artifact
 	}
 
 	names := []string{artifact.Id()}
-	tags := artifact.State("docker_tags")
-	if tags != nil {
-		cast := tags.([]interface{})
-		for _, name := range cast {
-			if n, ok := name.(string); ok {
-				names = append(names, n)
-			}
+	tags, _ := artifact.State("docker_tags").([]interface{})
```

Comment:
is it possible here that `[]interface` becomes a `[]string` in some cases ?

Reply:
Because in the above example they are

Reply:
I got so confused with these two types 🤦   I don't know why this part docker_tags is `[]interface{}` I first implement the obvious which was `[]string` but it didn't work, it expects to be `[]interface{}` here. 
The places docker_tags is set are:
- [here](https://github.com/hashicorp/packer/blob/master/post-processor/docker-push/post-processor.go#L129) - with type [interface{}](https://github.com/hashicorp/packer/blob/master/post-processor/docker-push/post-processor.go#L107) (returned by `artifact.State`) with nil or a slice of strings value
- [here](https://github.com/hashicorp/packer/blob/master/post-processor/docker-tag/post-processor.go#L115) - with type [[]string](https://github.com/hashicorp/packer/blob/master/post-processor/docker-tag/post-processor.go#L85)

If docker-push is used without docket-tag, tags will be nil before and after. 
If docker-tag is used before, tags will be set as []string.
No idea why here is []interface{}

Reply:
I tested all combinations and all of them failed here for `[]string` and only worked for `[]interface{}`

Reply:
https://github.com/hashicorp/packer/issues/9995 the reporter even tested the version where I changed to `[]string` and it broke the logic

Reply:
I might be missing some information somewhere